net/types.go

```diff
@@ -27,16 +29,21 @@ type (
 // implemented by *xmrmaker.Instance.
 type MakerHandler interface {
 	GetOffers() []*types.Offer
-	HandleInitiateMessage(msg *SendKeysMessage) (SwapState, Message, error)
+	HandleInitiateMessage(peerID peer.ID, msg *SendKeysMessage) (SwapState, Message, error)
 }
 
-// TakerHandler handles relay claim requests. It is implemented by
-// *xmrtaker.xmrtaker.
-type TakerHandler interface {
+// RelayHandler handles relay claim requests. It is implemented by
+// *backend.backend.
+type RelayHandler interface {
 	HandleRelayClaimRequest(msg *RelayClaimRequest) (*RelayClaimResponse, error)
 }
 
 type swap struct {
 	swapState SwapState
 	stream    libp2pnetwork.Stream
+	// isTaker is true if we initiate the swap (created the outbound stream). If
+	// the value is false, it is not safe to assume that we are the maker, as
+	// the swap entry will get created before validation of the first received
+	// message.
```

Comment:
how would we not be the maker if this is false?

Reply:
I'll remove that message.  I thought we were creating the entry before processing the initial received message, but I just looked at the code and it is fully processing the first message before creating the entry and calling `handleProtocolStreamInner`.

Reply:
Verbiage about the maker completely removed.